src/components/tx-flow/flows/TokenTransfer/ReviewSpendingLimitTx.tsx

```diff
@@ -88,7 +90,7 @@ const ReviewSpendingLimitTx = ({
 
     try {
       await dispatchSpendingLimitTxExecution(txParams, txOptions, onboard, safe.chainId, safeAddress)
-
+      setTxFlow(undefined)
       onSubmit()
```

Comment:
We should call `onSubmit` before unmount.

```suggestion
      onSubmit()
      setTxFlow(undefined)
```